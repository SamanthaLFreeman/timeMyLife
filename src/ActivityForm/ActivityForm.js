import React, { Component } from 'react';

class ActivityForm extends Component {
  constructor() {
    super();
    this.state = {
      categories: '',
      name: '',
      mins: '',
      secs: ''
    }
  }

  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
  }

  handleSubmit = () => {
    let newActivity = {
      categories: '',
      name: '',
      mins: '',
      secs: ''
    }
    this.props.setCurrentActivity(newActivity)
  }

  render() {
    return (
      <form className='ActivityForm'>
        <h2>New Activity</h2>
        <p>Select a category:</p>
        <div className='categories'>
          <button>Study</button>
          <button>Meditate</button>
          <button>Exercise</button>
        </div>
        <div className='inputs'>
          <input
            placeholder='Name Activity'
            type='text'
            name='name'
            value={this.state.name}
            onChange={this.handleChange} />
          <input
            placeholder='Minutes'
            type='number'
            name='mins'
            value={this.state.mins}
            onChange={this.handleChange} />
          <input
            placeholder='Seconds'
            type='number'
            name='secs'
            value={this.state.secs}
            onChange={this.handleChange} />
          <button onClick={this.handleSubmit}>Start Activity</button>
        </div>
      </form>
    )
  }
}

export default ActivityForm;