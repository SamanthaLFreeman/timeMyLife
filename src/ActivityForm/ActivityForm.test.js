import React from 'react';
import ActivityForm from './ActivityForm';
import { shallow } from 'enzyme';

describe('ActivityForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ActivityForm />)
  })

  it('should match the snapshot with the correct data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should update the state when handle change is called', () => {
    const mockEventCategories = {
      target: {
        name: 'categories',
        value: 'Study'
      }
    };
    const mockEventName = {
      target: {
        name: 'name',
        value: 'Read'
      }
    };
    const mockEventMins = {
      target: {
        name: 'mins',
        value: '10'
      }
    };
    const mockEventSecs = {
      target: {
        name: 'secs',
        value: '30'
      }
    };
    const expectedCategory = 'Study';
    const expectedName = 'Read';
    const expectedMins = '10';
    const expectedSecs = '30';
    wrapper.instance().handleChange(mockEventCategories);
    wrapper.instance().handleChange(mockEventName);
    wrapper.instance().handleChange(mockEventMins);
    wrapper.instance().handleChange(mockEventSecs);
    expect(wrapper.state('categories')).toEqual(expectedCategory);
    expect(wrapper.state('name')).toEqual(expectedName);
    expect(wrapper.state('mins')).toEqual(expectedMins);
    expect(wrapper.state('secs')).toEqual(expectedSecs);
  });
})