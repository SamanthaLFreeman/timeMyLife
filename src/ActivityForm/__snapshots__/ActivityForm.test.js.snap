// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityForm should match the snapshot with the correct data passed through 1`] = `
<form
  className="ActivityForm"
>
  <h2>
    New Activity
  </h2>
  <p>
    Select a category:
  </p>
  <div
    className="categories"
  >
    <button>
      Study
    </button>
    <button>
      Meditate
    </button>
    <button>
      Exercise
    </button>
  </div>
  <div
    className="inputs"
  >
    <input
      name="name"
      onChange={[Function]}
      placeholder="Name Activity"
      type="text"
      value=""
    />
    <input
      name="mins"
      onChange={[Function]}
      placeholder="Minutes"
      type="number"
      value=""
    />
    <input
      name="secs"
      onChange={[Function]}
      placeholder="Seconds"
      type="number"
      value=""
    />
    <button
      onClick={[Function]}
    >
      Start Activity
    </button>
  </div>
</form>
`;
